homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,221 @@
+            if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
+                stream_options = {
+                    "rtsp_flags": "prefer_tcp",
+                    "stimeout": "5000000",
+                }
+            if rtsp_transport := info.get(CONF_RTSP_TRANSPORT):
+                stream_options[FFMPEG_OPTION_MAP[CONF_RTSP_TRANSPORT]] = rtsp_transport
+            _LOGGER.debug("Attempting to open stream %s", stream_source)
+            container = await self.hass.async_add_executor_job(
+                partial(
+                    av.open,
+                    stream_source,
+                    options=stream_options,
+                    timeout=SOURCE_TIMEOUT,
+                )
+            )
+            _ = container.streams.video[0]
+            return True, ""
+        # pylint: disable=c-extension-no-member
+        except av.error.FileNotFoundError:
+            return False, "stream_no_route_to_host"
+        except av.error.HTTPUnauthorizedError:  # pylint: disable=c-extension-no-member
+            return False, "stream_unauthorised"
+        except (KeyError, IndexError):
+            return False, "stream_novideo"
+        except PermissionError:
+            return False, "stream_not_permitted"
+        except OSError as err:
+            if "No route to host" in str(err):
+                return False, "stream_no_route_to_host"
+            if "Input/output error" in str(err):
+                return False, "stream_io_error"
+            raise err
+
+    @staticmethod
+    def build_schema(user_input):
+        """Create schema for camera config setup."""
+        spec = {
+            vol.Optional(CONF_NAME, default=user_input[CONF_NAME]): str,
+            vol.Optional(
+                CONF_STILL_IMAGE_URL,
+                description={
+                    "suggested_value": user_input.get(CONF_STILL_IMAGE_URL, "")
+                },
+            ): str,
+            vol.Optional(
+                CONF_STREAM_SOURCE,
+                description={"suggested_value": user_input.get(CONF_STREAM_SOURCE, "")},
+            ): str,
+            vol.Optional(
+                CONF_RTSP_TRANSPORT, default=user_input.get(CONF_RTSP_TRANSPORT)
+            ): vol.In([None, "tcp", "udp", "udp_multicast", "http"]),
+            vol.Optional(
+                CONF_AUTHENTICATION, default=user_input.get(CONF_AUTHENTICATION)
+            ): vol.In([HTTP_BASIC_AUTHENTICATION, HTTP_DIGEST_AUTHENTICATION]),
+            vol.Optional(
+                CONF_USERNAME,
+                description={"suggested_value": user_input.get(CONF_USERNAME, "")},
+            ): str,
+            vol.Optional(
+                CONF_PASSWORD,
+                description={"suggested_value": user_input.get(CONF_PASSWORD, "")},
+            ): str,
+            vol.Optional(
+                CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+                default=user_input.get(CONF_LIMIT_REFETCH_TO_URL_CHANGE),
+            ): bool,
```

Comment:
Please move this to an options flow

Reply:
@bdraco thank you for the review!

My plan was to do config flow later to keep this PR as small as possible (see item 4 [above](https://github.com/home-assistant/core/pull/52360#issuecomment-1003618056)).  I have had to do some complex rebases already over the last year which were a real pain.  

Happy to do an options flow as a smaller follow up PR.  Is it necessary to do them together?

Reply:
We should do the options flow at the same time since you can import the data directly into options from yaml.

Example https://github.com/home-assistant/core/pull/66607

Reply:
Ok, done.